### PR TITLE
[coqdep] mli cleanup, remove unused functions

### PR DESCRIPTION
**Pull request by ejgallego**

### CODE REVIEW

File: tools/coqdep_common.ml [hunk 1]
```diff
@@ -125,8 +125,8 @@ let mkknown () =
     with Not_found -> None
   in add, iter, search
 
-let add_ml_known, iter_ml_known, search_ml_known = mkknown ()
-let add_mli_known, iter_mli_known, search_mli_known = mkknown ()
+let add_ml_known, _iter_ml_known, search_ml_known = mkknown ()
+let add_mli_known, _iter_mli_known, search_mli_known = mkknown ()
```
Comment (SkySkimmer) [inline tools/coqdep_common.ml]: No need to keep the iter imo

File: tools/coqdep.ml [hunk 2]
```diff
@@ -159,8 +117,6 @@ let coqdep () =
   List.iter (fun (f,d) -> add_mli_known f d ".mli") !mliAccu;
   List.iter (fun (f,d) -> add_mllib_known f d ".mllib") !mllibAccu;
   List.iter (fun (f,suff,d) -> add_ml_known f d suff) !mlAccu;
-  warning_mult ".mli" iter_mli_known;
-  warning_mult ".ml" iter_ml_known;
```
Comment (SkySkimmer) [inline tools/coqdep.ml]: Is the warning not useful?

**Comment by SkySkimmer** [comment]
@ejgallego needs fixing btw

File: tools/coqdep.ml (hunk 2 above)
Comment (ejgallego) [reply tools/coqdep.ml]: I'm not sure, what do you think?

Comment (SkySkimmer) [reply tools/coqdep.ml]: I don't know when it happens exactly.

**Review by SkySkimmer — APPROVED** [review]

**Pull request merged**